napari/layers/image/image.py

```diff
@@ -692,6 +698,43 @@ def _set_view_slice(self):
             self.reset_contrast_limits()
             self._should_calc_clims = False
 
+    def _project_slice(self):
+        """
+        Project a thick nD slice into a single 2d or 3d image that can be passed to the GPU.
+
+        Projection is based on slice thickness and the current projection mode.
+        """
+        if self.slice_projection == ImageSliceProjection.NONE:
+            return self.data[self._slice_indices]
+
+        half_thickness = (
+            np.maximum(np.ceil(self._thickness).astype(int), 1) / 2
+        )
```

Comment:
no need to use numpy here since `self._thickness` will always have a small number of elements.

```suggestion
        half_thickness = tuple(max(ceil(t), 1) / 2 for t in self._thickness)
```

Reply:
see alternative option in next comment where this wouldn't be computed at all for the displayed dimensions